src/openms/include/OpenMS/ANALYSIS/ID/PeptideIndexing.h

```diff
@@ -677,8 +692,143 @@ namespace OpenMS
+                                                          {"reverse", {0,0}}, {"dev", {0,0}},
+                                                          {"iddecoy", {0,0}}, {"xxx", {0,0}},
+                                                          {"shuffle", {0,0}}, {"pseudo", {0,0}},
+                                                          {"random", {0,0}}};
+
+       // map case insensitive strings back to original case (as used in fasta)
+       CaseInsensitiveToCaseSensitiveDecoy decoy_case_sensitive;
+
+       int all_prefix_occur(0), all_suffix_occur(0), all_proteins_count(0);
+
+       const size_t PROTEIN_CACHE_SIZE = 4e5;
+
+       while (true)
+       {
+         proteins.cacheChunk(PROTEIN_CACHE_SIZE);
+         if (!proteins.activateCache()) break;
+         auto prot_count = (SignedSize)proteins.chunkSize();
+         all_proteins_count += prot_count;
+
+         {
+           for (SignedSize i = 0; i < prot_count; ++i)
+           {
+             String seq = proteins.chunkAt(i).identifier;
+             String seq_lower = seq;
+             seq_lower.toLower();
+
+             // remove all non-alphanumeric characters
+             seq_lower.erase(std::remove_if(seq_lower.begin(), seq_lower.end(),
+                                            [](char c) { return !std::isalpha(c); } ), seq_lower.end());
+
+             // search for case insensitive occurrence of the possible decoys
+             for (auto pair : decoy_count)
+             {
+               if (seq_lower.hasPrefix(pair.first))
+               {
+                 all_prefix_occur++;
+                 // count observed (ignoring case)
+                 decoy_count[pair.first].first++;
+                 // store observed (case sensitive)
+                 std::string seq_decoy = StringUtils::prefix(seq, pair.first.length());
+                 decoy_case_sensitive[pair.first] = seq_decoy;
+               }
+               else if (seq_lower.hasSuffix(pair.first))
+               {
+                 all_suffix_occur++;
+                 // count observed (ignoring case)
+                 decoy_count[pair.first].second++;
+                 // store observed (case sensitive)
+                 std::string seq_decoy = StringUtils::suffix(seq, pair.first.length());
+                 decoy_case_sensitive[pair.first] = seq_decoy;
+               }
+             }
+           }
+         }
+       }
+
+       // DEBUG only: print counts of found decoys
+       for (auto &a : decoy_count) LOG_DEBUG << a.first << "\t" << a.second.first << "\t" << a.second.second << std::endl;
+
+       if (all_prefix_occur == all_suffix_occur)
+       {
+         LOG_ERROR << "Unable to determine decoy string!" << std::endl;
+         return false;
+       }
+
+       // side case: prefix_reverse == prefix_rev -> if reverse occurs, then rev will be found as well -> remove rev findings
+       if (decoy_count["rev"].first == decoy_count["reverse"].first)
```

Comment:
the `==` might break. Imagine the protein `REV1_HUMAN` (yes, it exists).
the safest option would be to include the underscores, e.g. `rev_`, which would make it unique, but I guess one cannot rely on underscores being used?!
What if you check for `reverse` first and if its found, stop iterating over the other affixes? Then you'd have at most a few spurious hits for `rev`, but`reverse` would still win if its truly present and you dont need to handle the special case any further.

Reply:
This sidecased has been removed. 
See: https://github.com/OpenMS/OpenMS/pull/3454/commits/c4f95fbef454a9c548811bbd737f8fc6ba7f98dc